prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/CustomFHIRFunctions.kt

```diff
@@ -207,11 +212,24 @@ object CustomFHIRFunctions : FhirPathFunctions {
         }
 
         var timePeriodSpecified: String? = null
-        var comparisonDate: LocalDate = LocalDate.now()
-        if (!parameters.isNullOrEmpty() && !parameters[0].isNullOrEmpty()) {
-            val params = getParamValue(parameters[0])
-            timePeriodSpecified = params.second
-            comparisonDate = params.first
+        var comparisonDate: LocalDate? = LocalDate.now()
+        if (!parameters.isNullOrEmpty()) {
+            if (parameters.size > 2) {
+                throw SchemaException("Must call the convertDateToAge function with no more than two parameters")
+            }
+
+            parameters.forEach {
+                it.forEach { actualParam ->
```

Comment:
Instead of looping through the sublist with another `forEach`, we should probably instead check that each sublist has exactly 1 element and just throw another SchemaException if that's not the case.

Reply:
If you take the suggestion from the other comment, this can actually just become something like `val actualParam = it[0]` since it will be validated above.

Reply:
So, after changing this and having the tests fail, I remembered that it is just a list of a list that holds the params so listOf(listOf(param1, param2)) so we actually need that second foreach because that is where the actual params are. I guess I could do the check for the first list, but given that is just how it works, it seems odd